acceptance/upgrade/upgrade_bound_test.go

```diff
@@ -0,0 +1,179 @@
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("<Upgrade2> Epinio upgrade with bound app and services", func() {
+	var (
+		namespace string // Namespace created before upgrade
+		appName   string // Application created before upgrade
+		service   string // Service created after upgrade
+
+		epinioHelper epinio.Epinio
+	)
+
+	BeforeEach(func() {
+		epinioHelper = epinio.NewEpinioHelper(testenv.EpinioBinaryPath())
+
+		// Before upgrade ...
+		namespace = catalog.NewNamespaceName()
+		env.SetupAndTargetNamespace(namespace)
+		appName = catalog.NewAppName()
+		service = catalog.NewServiceName()
+	})
+
+	AfterEach(func() {
+		if !CurrentSpecReport().Failed() {
+			// After upgrade ...
+			env.DeleteApp(appName)
+			env.DeleteService(service)
+			env.DeleteNamespace(namespace)
+		}
+	})
+
+	It("Can upgrade epinio bound to a custom service", func() {
+
+		// Note current versions of client and server
+		By("Versions before upgrade")
+		env.Versions()
+
+		// Create a service
+		By("Creating a service")
+		out, err := env.Epinio("", "service", "create", "mysql-dev", service)
+		Expect(err).ToNot(HaveOccurred(), out)
+
+		// Debug only
+		fmt.Fprintf(GinkgoWriter, "Service create log: %v\n", out)
+
+		By("Wait for deployment")
+		// Make outValue global variable as we'll need it later
+		var outValue string
+		Eventually(func() string {
+			outValue, _ = env.Epinio("", "service", "show", service)
+			return outValue
+		}, "2m", "5s").Should(
+			HaveATable(
+				WithHeaders("KEY", "VALUE"),
+				WithRow("Status", "deployed"),
+			),
+		)
+		// store the service route
+		svcRouteRegexp := regexp.MustCompile(`\b(\w{29}-mysql)`)
+		svcRoute := string(svcRouteRegexp.Find([]byte(outValue)))
+		fmt.Fprintf(GinkgoWriter, "The service table content: %v\n", outValue)
+		fmt.Fprintf(GinkgoWriter, "The service route is: %v\n", svcRoute)
+
+		// Deploy Wordpress application
+		By("Pushing Wordpress App")
+		wordpress := "https://github.com/epinio/example-wordpress,main"
+
+		pushLog, err := env.EpinioPush("",
+			appName,
+			"--name", appName,
+			"--git", wordpress,
+			"-e", "BP_PHP_WEB_DIR=wordpress",
+			"-e", "BP_PHP_VERSION=8.0.x",
+			"-e", "BP_PHP_SERVER=nginx",
+			"-e", "DB_HOST="+svcRoute,
+			"-e", "SERVICE_NAME="+service)
+		Expect(err).ToNot(HaveOccurred(), pushLog)
+
+		routeRegexp := regexp.MustCompile(`https:\/\/.*omg.howdoi.website`)
+		route := string(routeRegexp.Find([]byte(pushLog)))
+
+		Eventually(func() string {
+			out, err := env.Epinio("", "app", "list")
+			Expect(err).ToNot(HaveOccurred(), out)
+			return out
+		}, "5m").Should(
+			HaveATable(
+				WithHeaders("NAME", "CREATED", "STATUS", "ROUTES", "CONFIGURATIONS", "STATUS DETAILS"),
+				WithRow(appName, WithDate(), "1/1", appName+".*", "", ""),
+			),
+		)
+
+		By("Bind it")
+		out, err = env.Epinio("", "service", "bind", service, appName)
+		Expect(err).ToNot(HaveOccurred(), out)
+		// This should be done as Eventually block
```

Comment:
please remove this comment, it is one shot command so it makes no sense to use Eventually here